clients/src/main/java/org/apache/kafka/clients/admin/internals/DescribeShareGroupsHandler.java

```diff
@@ -0,0 +1,381 @@
+    final Set<CoordinatorKey> groupsToUnmap = new HashSet<>();
+
+    if (abstractResponse instanceof DescribeGroupsResponse) {
+      return handledClassicGroupResponse(
+          coordinator,
+          completed,
+          failed,
+          groupsToUnmap,
+          (DescribeGroupsResponse) abstractResponse
+      );
+    } else if (abstractResponse instanceof ShareGroupDescribeResponse) {
+      return handledShareGroupResponse(
+          coordinator,
+          completed,
+          failed,
+          groupsToUnmap,
+          (ShareGroupDescribeResponse) abstractResponse
+      );
+    } else {
+      throw new IllegalArgumentException("Received an unexpected response type.");
+    }
+  }
+
+  @Override
+  public Map<CoordinatorKey, Throwable> handleUnsupportedVersionException(
+      int coordinator,
+      UnsupportedVersionException exception,
+      Set<CoordinatorKey> keys
+  ) {
+    Map<CoordinatorKey, Throwable> errors = new HashMap<>();
+
+    keys.forEach(key -> {
+      if (!useClassicGroupApi.add(key.idValue)) {
+        // We already tried with the classic group API so we need to fail now.
+        errors.put(key, exception);
+      }
+    });
+
+    return errors;
+  }
+
+  private ApiResult<CoordinatorKey, ShareGroupDescription> handledShareGroupResponse(
+      Node coordinator,
+      Map<CoordinatorKey, ShareGroupDescription> completed,
+      Map<CoordinatorKey, Throwable> failed,
+      Set<CoordinatorKey> groupsToUnmap,
+      ShareGroupDescribeResponse response
+  ) {
+    for (ShareGroupDescribeResponseData.DescribedGroup describedGroup : response.data().groups()) {
+      final CoordinatorKey groupIdKey = CoordinatorKey.byGroupId(describedGroup.groupId());
+      final Errors error = Errors.forCode(describedGroup.errorCode());
+      if (error != Errors.NONE) {
+        handleError(
+            groupIdKey,
+            error,
+            describedGroup.errorMessage(),
+            failed,
+            groupsToUnmap,
+            true
+        );
+        continue;
+      }
+
+      final Set<AclOperation> authorizedOperations = validAclOperations(describedGroup.authorizedOperations());
+      final List<MemberDescription> memberDescriptions = new ArrayList<>(describedGroup.members().size());
+
+      describedGroup.members().forEach(groupMember -> {
+        memberDescriptions.add(new MemberDescription(
+            groupMember.memberId(),
+            Optional.ofNullable(groupMember.instanceId()),
+            groupMember.clientId(),
+            groupMember.clientHost(),
+            new MemberAssignment(convertAssignment(groupMember.assignment())),
+            Optional.of(new MemberAssignment(convertAssignment(groupMember.assignment())))
+        ));
+      });
+
+      final ShareGroupDescription shareGroupDescription =
+          new ShareGroupDescription(
+              groupIdKey.idValue,
+              memberDescriptions,
+              describedGroup.assignorName(),
+              GroupType.CONSUMER,
```

Comment:
`GroupType.SHARE`

Reply:
> GroupType.SHARE

@AndrewJSchofield 
As per the comments above, have updated the class `ShareGroupDescription` by removing group type and partition assignor field. Hence, the GroupType argument is not needed.